src/app/approve/approve.component.ts

```diff
@@ -369,7 +369,7 @@ export class ApproveComponent implements OnInit {
   toFixedLengthDecimalString(num: number): string {
     // Change nanos into a formatted string of units. This combination of toFixed and regex removes trailing zeros.
     // If we do a regular toString(), some numbers can be represented in E notation which doesn't look as good.
-    return num.toFixed(9).replace(/([0-9]+(\.[0-9]+[1-9])?)(\.?0+$)/, '$1');
+    return num.toFixed(9).replace(/^(\d*\.\d*?[1-9])0+$/, '$1');
```

Comment:
Trying a few examples in console, it doesn't look like this gives us exactly what we want when dealing with integers:

Ex:
```
(10).toFixed(9).replace(/^(\d*\.\d*?[1-9])0+$/, '$1')
'10.000000000'
```

We should only apply the regex if the number is not an integer.